### PR TITLE
Agdroid 504 generic store config

### PR DESCRIPTION
**Pull request by secondsun**
@matzew :eyes:?

This covers AGDROID-504 and a small update to the behavior of opening SQL based stores.

### CODE REVIEW

File: aerogear-android-store/src/main/java/org/jboss/aerogear/android/store/sql/EncryptedSQLStore.java [hunk 1]
```diff
@@ -351,7 +343,8 @@ private boolean isOpen() {
 
     private void ensureOpen() {
         if (!isOpen()) {
-            throw new StoreNotOpenException();
+            Log.w(TAG, "Store is not opened, trying to open.");
```
Comment (matzew) [inline aerogear-android-store/src/main/java/org/jboss/aerogear/android/store/sql/EncryptedSQLStore.java]: I do like this

**Comment by matzew** [comment]
@secondsun this will make Travis happy: https://github.com/secondsun/aerogear-android-store/pull/3/files

**Comment by matzew** [comment]
@secondsun looks like you did change this similar to my PR :)

**Comment by secondsun** [comment]
@matzew I think we were both fixing the same bug :)

**Comment by matzew** [comment]
👍

**Pull request merged**